### PR TITLE
📖 Add deletion strategy ADR

### LINKED ISSUE #63: ADR: Deletion strategy
### What would you like to be added (User Story)?

As a developer
I want an ADR to document our deletion strategy
So future engineers understand the decision

### Detailed Description

We need an ADR to document the decision around how we handle _deletion.

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind feature

### PR DESCRIPTION
**Pull request by furkatgofurov7**
kind/documentation

**What this PR does / why we need it**:
Adds deletion strategy ADR

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63 

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits into logical changes
- [X] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: docs/adr/0003-deletion-strategy.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+**Cluster Annotation:** When deleting a Rancher cluster, the operator will annotate the corresponding CAPI cluster with the `ClusterImportedAnnotation` annotation. This annotation will prevent automatic re-import of the CAPI cluster, only if the Rancher cluster itself contains the same annotation.
+
+From an end-user perspective:
+
+* If user manually deletes CAPI cluster. Rancher turtles uses [Kubernetes owner references](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) to track relationships between objects. These references are used for Kubernetes garbage collection, which is the basis of Rancher cluster deletion in Rancher turtles.
+* If user manually removes the Rancher cluster from UI. Depending on annotation presence in CAPI cluster, operator decides to allow or block importing. 
+
+## Consequences
+
+- The operator will leverage Kubernetes' native garbage collection mechanism, by using the owner reference chain, to ensure a streamlined cluster deletion process.
```
Comment (Danil-Grigorev) [inline docs/adr/0003-deletion-strategy.md:27]: Not sure if it is will be possible when running rancher turtles outside of rancher manager cluster, we will have to find a suitable alternative later.

File: docs/adr/0003-deletion-strategy.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+As operator, it is crucial to establish clear guidelines for how Rancher turtles handle cluster deletion operations. The objective is to facilitate the removal of a CAPI-imported cluster from Rancher, ensuring that it no longer appears in the Rancher UI, all while keeping the underlying CAPI cluster intact.
+
+## Decision
+
+The deletion workflow is structured as follows:
+
+**Ownership Chain:** Rancher cluster is associated with CAPI cluster through the owner references chain during their creation process.
+
+**Cluster Annotation:** When deleting a Rancher cluster, the operator will annotate the corresponding CAPI cluster with the `ClusterImportedAnnotation` annotation. This annotation will prevent automatic re-import of the CAPI cluster, only if the Rancher cluster itself contains the same annotation.
```
Comment (Danil-Grigorev) [inline docs/adr/0003-deletion-strategy.md:18]: From the name of the annotation it is not clear what is it actually set on the object. People should not have to look in the code after reading this ADR for an actual value. Might be better to also mention it is `imported` annotation here.

File: docs/adr/0003-deletion-strategy.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+From an end-user perspective:
+
+* If user manually deletes CAPI cluster. Rancher turtles uses [Kubernetes owner references](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) to track relationships between objects. These references are used for Kubernetes garbage collection, which is the basis of Rancher cluster deletion in Rancher turtles.
+* If user manually removes the Rancher cluster from UI. Depending on annotation presence in CAPI cluster, operator decides to allow or block importing.
```
Comment (Danil-Grigorev) [inline docs/adr/0003-deletion-strategy.md:23]:
```suggestion
* If user manually removes the Rancher cluster from UI: depending on the `imported` annotation presence in the CAPI cluster, operator blocks Rancher cluster from being re-imported (re-created) after deletion.
```

File: docs/adr/0003-deletion-strategy.md (hunk 2 above)
Comment (Danil-Grigorev) [inline docs/adr/0003-deletion-strategy.md:18]:
```suggestion
**Cluster Annotation:** When deleting a Rancher cluster, the operator will annotate the corresponding CAPI cluster with the `ClusterImportedAnnotation` (`imported=“true”`) annotation. This annotation will prevent automatic re-import of the CAPI cluster after corresponding Rancher cluster deletion. The underlying infrastructure provisioned by CAPI is left intact.
```

File: docs/adr/0003-deletion-strategy.md (hunk 1 above)
Comment (furkatgofurov7) [reply docs/adr/0003-deletion-strategy.md:27]: This ADR is based on the current state of the operator, we would need to update it probably accordingly later when we can run rancher turtles outside of rancher manager cluster

File: docs/adr/0003-deletion-strategy.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## Context
+
+As operator, it is crucial to establish clear guidelines for how Rancher turtles handle cluster deletion operations. The objective is to facilitate the removal of a CAPI-imported cluster from Rancher, ensuring that it no longer appears in the Rancher UI, all while keeping the underlying CAPI cluster intact.
+
+## Decision
+
+The deletion workflow is structured as follows:
+
+**Ownership Chain:** Rancher cluster is associated with CAPI cluster through the owner references chain during their creation process.
```
Comment (Danil-Grigorev) [inline docs/adr/0003-deletion-strategy.md:16]:
```suggestion
**Ownership Chain:** Rancher cluster is associated with CAPI cluster through the owner references chain during their creation process. CAPI cluster owns Rancher cluster.
```
One more small detail.

Comment (furkatgofurov7) [reply docs/adr/0003-deletion-strategy.md]: Thanks, updated now

**Review by Danil-Grigorev — APPROVED** [review]

**Review by alexander-demicev — APPROVED** [review]

**Review by mjura — APPROVED** [review]
lgtm

**Pull request merged**